x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/Transform.java

```diff
@@ -377,6 +380,35 @@ public void close() {
         return List.of(AUDIT_INDEX_PATTERN);
```

Comment:
@gwbrown 

- Is this + the `getSystemIndexDescriptors` the list of indexes to be deleted at cleanup?
- Is this used _only_ for deletion or for something else, too?

There is 1 case to be aware of, if a user started with an older version, used transform on it and upgraded the installation once or more, he has older indices still lying around. Those must be deleted at reset as well. 

It seems to me, this is an overlook introduced earlier(this PR is not to blame). Let me know, I can create a tiny separate PR once I have clarification from you.

With other words this should return `AUDIT_INDEX_PATTERN + INDEX_NAME_PATTERN_DEPRECATED + .transform-internal-00{1-5}` (with `.transform-internal-006` we switched over to a system index)

Reply:
1. Yes, this and `getSystemIndexDescriptors` are used for deletion.
2. Associated indices are also automatically included in a snapshot if the plugin's feature state (that is, system indices) are included in the snapshot, to ensure the snapshot can be used to restore all data owned by the feature. That's the only other use of associated indices at this time, though (and we don't have immediate plans for more, but there's always possibilities).

It sounds like yes, we should include the deprecated patterns in the associated indices. The system index descriptor already uses the `.transform-internal-*` indices:
https://github.com/elastic/elasticsearch/blob/54ed2e37d9fc5432c9c76ab05cded8d11a64133e/x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/persistence/TransformInternalIndex.java#L90
So those will be picked up as part of the system indices - they don't need to (and shouldn't) be registered as associated indices as well.

Reply:
ok, thanks. That means the only uncovered one is `TransformInternalIndexConstants.INDEX_NAME_PATTERN_DEPRECATED` (needed only for the `7.x` series, for `8` we plan to provide an upgrader that gets rid of the deprecated indexes)